feat: Add map share rpc protobufs

RangerMauve opened the pull request.

gmaclennan commented on proto/extensions.proto:
```diff
@@ -32,3 +32,28 @@ message DownloadIntentExtension {
+  // The ID of the map being shared 
+  string mapId = 6;
+  // The timestamp when the map share invite was received
+  uint64 receivedAt = 7;
```
Not needed in the extension message - this would be set by the receiver.

gmaclennan commented on proto/extensions.proto:
```diff
@@ -32,3 +32,28 @@ message DownloadIntentExtension {
+  // The minimum zoom level of the map data being shared 
+  int32 minzoom = 9;
+  // The maximum zoom level of the map data being shared
+  int32 maxzoom = 10;
```
These will need an additional validation step - bounds must be a tuple of length 4 within these max: `[-180, -85.051129, 180, 85.051129]` (that's the max for spherical mercator projection)

Zoom must be between 0-22.

RangerMauve commented on proto/extensions.proto:
```diff
@@ -32,3 +32,28 @@ message DownloadIntentExtension {
+  int32 maxzoom = 10;
+  // Estimated size of the map data being shared in bytes 
+  uint64 estimatedSizeBytes = 11;
+}
```
What does the created property represent?

gmaclennan replied:
the timestamp of when the map was created. I've renamed this `mapCreated` in the server code, so maybe use that here too. I've also added a `mapShareCreated` field - when the map share was originally created.

RangerMauve commented:
@gmaclennan Made changes

- Updated fields based on your feedback (created fields, sender device id and name, download/decline URLs)
- Added validation for share extension
- Added `map-share-error` event for detecting failed map share attempts (just logging for now)
- Added tests for validation of map share extension
- Validating map share *before* it's sent down the extension (with bypass for tests)
- Ignoring map shares from blocked/left members
- Added e2e tests to check that the share is being received and that validation is happening
